core/chains/evm/txmgr/attempts.go

```diff
@@ -119,9 +119,17 @@ func (c *evmTxAttemptBuilder) NewEmptyTxAttempt(nonce evmtypes.Nonce, feeLimit u
 		return attempt, errors.New("NewEmptyTranscation: legacy fee cannot be nil")
 	}
 
-	tx := types.NewTransaction(uint64(nonce), fromAddress, value, uint64(feeLimit), fee.Legacy.ToInt(), payload)
+	tx := newLegacyTransaction(
```

Comment:
TBH this feels redundant. We could just create a new `types.LegacyTx ` instead of having a function that just does 2 casts on the params. `newLegacyTransaction` is not that frequent anyway, we use it only twice.